Dockerfile

```diff
@@ -9,129 +9,122 @@ ENV DEBIAN_FRONTEND=noninteractive \
-        libxcursor1 \
-        libxi6 \
-        libxext6 \
-        libxfixes3 \
-        zlib1g \
-        libgl1 \
-        pulseaudio \
-        socat \
+    apt-transport-https \
+    autoconf \
+    autotools-dev \
+    build-essential \
+    curl \
+    file \
+    gcc \
+    git \
+    g++ \
+    gnupg2 \
+    lcov \
+    libc++1-10 \
+    libgl1 \
+    libglu1-mesa \
+    libtcmalloc-minimal4 \
+    libtool \
+    locales \
+    make \
+    openjdk-11-jdk-headless \
+    openssh-client \
+    patch \
+    python2 \
+    python3 \
+    python3-distutils \
+    rsync \
+    ruby \
+    ruby-dev \
+    tzdata \
+    unzip \
+    sudo \
+    ninja-build \
+    wget \
+    zip \
+    # Dev libraries requested by Hermes
+    libicu-dev \
+    # Emulator & video bridge dependencies
+    libc6 \
+    libdbus-1-3 \
+    libfontconfig1 \
+    libgcc1 \
+    libpulse0 \
+    libtinfo5 \
+    libx11-6 \
+    libxcb1 \
+    libxdamage1 \
+    libnss3 \
+    libxcomposite1 \
+    libxcursor1 \
+    libxi6 \
+    libxext6 \
+    libxfixes3 \
+    zlib1g \
+    libgl1 \
+    pulseaudio \
+    socat \
     # for x86 emulators
     && apt-get install -qq -y \
-      libxtst6 \
-      libnspr4 \
-      libxss1 \
-      libasound2 \
-      libatk-bridge2.0-0 \
-      libgtk-3-0 \
-      libgdk-pixbuf2.0-0 \
+    libxtst6 \
+    libnspr4 \
+    libxss1 \
+    libasound2 \
+    libatk-bridge2.0-0 \
+    libgtk-3-0 \
+    libgdk-pixbuf2.0-0 \
     # python link required for NDK toolchain build to work
     && ln -s python3 /usr/bin/python \
     && gem install bundler \
     && echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers \
     && sh -c 'echo "en_US.UTF-8 UTF-8" > /etc/locale.gen' \
     && locale-gen \
-    && update-locale LANG=en_US.UTF-8 \
-    # install nodejs using n
-    && curl -L https://raw.githubusercontent.com/tj/n/master/bin/n -o n \
+    && update-locale LANG=en_US.UTF-8 
+
+# install nodejs using n
+RUN curl -L https://raw.githubusercontent.com/tj/n/master/bin/n -o n \
     && bash n $NODE_VERSION \
     && rm n \
     && npm install -g n \
-    && npm install -g yarn \
-    # download and install buck using the java11 pex from Jitpack
-    && curl -L https://jitpack.io/com/github/facebook/buck/v${BUCK_VERSION}/buck-v${BUCK_VERSION}-java11.pex -o /tmp/buck.pex \
-    && mv /tmp/buck.pex /usr/local/bin/buck \
-    && chmod +x /usr/local/bin/buck \
-    # download and unpack android
-    && wget -q https://dl.google.com/android/repository/${SDK_VERSION} -O /tmp/sdk.zip \
+    && npm install -g yarn 
+
+# download and unpack android
+RUN curl -sS https://dl.google.com/android/repository/${SDK_VERSION} -o /tmp/sdk.zip \
     && mkdir -p ${ANDROID_HOME}/cmdline-tools \
     && unzip -q -d ${ANDROID_HOME}/cmdline-tools /tmp/sdk.zip \
     && mv ${ANDROID_HOME}/cmdline-tools/cmdline-tools ${ANDROID_HOME}/cmdline-tools/latest \
     && rm /tmp/sdk.zip \
-    && mkdir -p /root/.android \
-    && touch /root/.android/repositories.cfg \
```

Comment:
This used to be necessary for avoiding some kind of user interaction requirement when installing / running for the first time. However if this runs ok without this then I guess no need for it any more.

Reply:
yeah it seems to have work based on the CI, unless you're referring to a different part of the process 🤷‍♂️

Reply:
think you're onto something here. attempted to start the emulator using:

```console
$ANDROID_HOME/emulator/emulator -avd emu -no-window -gpu swiftshader_indirect -no-snapshot -noaudio -no-boot-anim -camera-back emulated
```

and noticed the following warning:

```console
WARNING | Failed to open /root/.android/adbkey
WARNING | adbkey generation failed
```

not sure if this causes any noticeable issue, but warning goes away if adding the `mkdir /root/.android` command back, so maybe better to include anyways just in case.

Reply:
restored both commands via https://github.com/digidem/docker-android/pull/3/commits/7359bbcd29f226c83d1b09ea64f08ac59915bfde

Reply:
Great. ADB is used by detox I think to communicate with the app during tests (at least to set up tcp forwarding). Re hardware acceleration, you can't actually run the emulator in a standard docker container without KVM. The reason we use cirrus is that it is one of the few CIs that has enabled kvm and allows us to run the x86 emulator.